zerobuf/Zerobuf.cpp

```diff
@@ -6,61 +6,327 @@
 #include "Zerobuf.h"
 
 #include "detail/JsonSerialization.h"
-#include "Allocator.h"
+#include "ConstNonMovingSubAllocator.h"
+#include "NonMovingSubAllocator.h"
 #include "Schema.h"
+#include "Vector.h"
+#include "ConstVector.h"
 
 #include <iostream>
 
-
 namespace zerobuf
 {
 
+namespace detail
+{
+
+template<>
+void convertStaticToJSONValue<Zerobuf>( const Allocator* allocator,
+                                        const Schema::Field& valueSchema,
+                                        Json::Value& value )
+{
+    const size_t offset =
+        std::get< Schema::FIELD_DATAOFFSET >( valueSchema );
+    const size_t size = std::get< Schema::FIELD_SIZE >( valueSchema );
+    const Schema::SchemaFunction& schemaFunc =
+            std::get< Schema::FIELD_SCHEMAFUNC >( valueSchema );
+
+    const Schema& schema = schemaFunc();
+    const size_t staticSize = schema.staticSize;
+
+    const NonMovingBaseAllocator* parentAllocator =
+               static_cast< const NonMovingBaseAllocator* >( allocator );
+
+    if( size == 0 )
+    {
+        ConstNonMovingSubAllocator nonMovingAllocator( parentAllocator,
+                                                       offset,
+                                                       0,
+                                                       schema.staticSize );
+
+        detail::toJSONValue( &nonMovingAllocator, schema, value );
+
+    }
+    else /* static array */
+    if( std::get< Schema::FIELD_ISSTATIC >( valueSchema ))
+    {
+        for( size_t i = 0; i < size; ++i )
+        {
+            ConstNonMovingSubAllocator nonMovingAllocator( parentAllocator,
+                                                           offset + i * staticSize,
+                                                           0,
+                                                           schema.staticSize );
+            Json::Value jsonVal;
+            detail::toJSONValue( &nonMovingAllocator, schema, jsonVal );
+            value.append( jsonVal );
+        }
+    }
+}
+
+template<>
+void convertToJSONValue<Zerobuf>( const Allocator* allocator,
+                                  const Schema::Field& valueSchema,
+                                  Json::Value& value )
+{
+    if( std::get< Schema::FIELD_ISSTATIC >( valueSchema ))
+       convertStaticToJSONValue<Zerobuf>( allocator, valueSchema, value );
+    else /* dynamic variables */
+    {
+        const size_t offset =
+            std::get< Schema::FIELD_DATAOFFSET >( valueSchema );
+
+        const Schema::SchemaFunction& schemaFunc =
+                std::get< Schema::FIELD_SCHEMAFUNC >( valueSchema );
+
+        const Schema& schema = schemaFunc();
+
+        const NonMovingBaseAllocator* parentAllocator =
+                   static_cast< const NonMovingBaseAllocator* >( allocator );
+
+        ConstVector< Zerobuf > values( parentAllocator, offset, schema.staticSize );
```

Comment:
That only works by pure luck. Zerobuf is pure virtual, and can't be allocated. You really only need the number of elements, can't you get that somewhere else?